lib/iris/coords.py

```diff
@@ -977,12 +1050,13 @@ def serialize(x):
                 warnings.warn(msg.format(self.name()))
 
             # Create bounds for the new collapsed coordinate.
-            item = self.bounds if self.bounds is not None else self.points
-            lower, upper = np.min(item), np.max(item)
+            item = self.core_bounds() if self.has_bounds() \
+                else self.core_points()
+            lower, upper = item.min(), item.max()
             bounds_dtype = item.dtype
             bounds = [lower, upper]
```

Comment:
This line is actually a bit broken now, in that you can't successfully create a Coord like "AuxCoord(... bounds=[lazy_lower, lazy_upper])", as that argument is neither lazy or convertible to an array (it just fails, I think).

I see this breaks no tests.  It will require a collapse on a lazy coord to trigger it, which can only be loaded from NetCDF, and we probably don't have such a case.

There are bigger pre-existing problems here though, which I would say :
  * (a) the construction [lower, upper] is actually making a shape "(2,)+points.shape", when it **should** be ""points.shape+(2,)".  So this code actually only works for 1-D coords.
  * (b) thinking a bit more, I don't think this [min, max] calculation is actually sensible for multidimensional coords.  CF isn't really clear on the meaning of bounds on multidimensional coords, so we should probably just not produce result bounds in that case.  See : http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html#cell-boundaries
  * (c) likewise, we could have nbounds > 2 and again we have no clear interpretation, so again the result should not have bounds.